tests/tests_mqtt_server/src/server.rs

```diff
@@ -0,0 +1,68 @@
+use once_cell::sync::OnceCell;
+use std::{
+    net::TcpStream,
+    process::{Child, Command, Stdio},
+    sync::atomic::{AtomicUsize, Ordering},
+    thread::sleep,
+    time::Duration,
+};
+
+// Child doesn't implement `Drop` therefore we have to shutdown the process by hand.
+// Use of `static mut` requires some unsafe code.
+static mut SERVER: OnceCell<Child> = OnceCell::new();
+static CALLERS_COUNTER: AtomicUsize = AtomicUsize::new(0);
+
+fn start_server(port: u16) {
+    // `unsafe` has to be used due to `SERVER` being `static mut`.
+    unsafe {
+        let _server_child_handle = SERVER.get_or_init(|| spawn_server_process(port));
+    }
+
+    if TcpStream::connect(("127.0.0.1", port)).is_err() {
+        sleep(Duration::from_millis(100));
+    };
+
+    CALLERS_COUNTER.fetch_add(1, Ordering::Relaxed);
+}
+
+fn try_stop_server() {
+    CALLERS_COUNTER.fetch_sub(1, Ordering::Relaxed);
+
+    if CALLERS_COUNTER.load(Ordering::Relaxed) == 0 {
+        // `unsafe` has to be used due to `SERVER` being `static mut`,
+        // race between callers may occur in rare cases when a late caller will queue for the atomic operation after previous caller brought the count to 0.
+        unsafe {
+            // TODO: Remove unwrap.
+            let server_child_handle = SERVER.get_mut().unwrap();
+
+            // TODO: Use the result.
+            let _ = server_child_handle.kill();
+        }
+    }
+}
```

Comment:
We are looking here for the active connections before we kill it right?
So, we can do it this way "mosquitto_sub  -t '$SYS/broker/clients/connected'"